types/sinon/sinon-tests.ts

```diff
@@ -830,6 +830,11 @@ function testTypedStub() {
     const fooStub: sinon.SinonStubbedInstance<Foo> = {
         bar: sinon.stub(),
     };
+
+    stub.firstCall.args[0] = 42;
+
+    const stub3 = sinon.stub<readonly [number, string], boolean>();
+    stub3.firstCall.args[0] = 42; // $ExpectError
```

Comment:
since nobody should ever really want to assign an arg, it may be better to just test the type of the stub with an ExpectType comment

Reply:
I agree. It is probably also easier to understand the intent of the test-case when we check the type explicitly.